plugins/generator-1.18.2/forge-1.18.2/templates/gui/gui_container.java.ftl

```diff
@@ -144,8 +144,9 @@ public class ${name}Menu extends AbstractContainerMenu implements Supplier<Map<I
 						}
         	        <#elseif component.getClass().getSimpleName() == "InputSlot">
 						<#if component.inputLimit.toString()?has_content>
-        	             @Override public boolean mayPlace(ItemStack stack) {
-							 return (${mappedMCItemToItem(component.inputLimit)} == stack.getItem());
+        	             @Override public boolean mayPlace(ItemStack itemstack) {
+							 return (${mappedMCItemToItemOrTag(component.inputLimit, "ITEM")}

```

Comment:
This is misleading naming as calling `mappedMCItemToItemOrTag` will give you either item or boolean for tag, assuming itemstack as variable name. I don't think this belongs to mcitems, at least the implementation you made as it makes unclean and unclear code

Reply:
I was not sure how to make it. Do you have a better idea?

Reply:
Make the code inline here maybe, as delegating checking equality to mcitems that are meant for mapping mcitems is not ok.

Reply:
This also applies to 1.18.x

Reply:
I know. I didn't have the time to finish and test.